### ISSUE TITLE
Jdaviz is hard-to-read and partly broken in Jupyterlab dark mode

### ISSUE THREAD

**Issue by eteq** [spacetelescope/jdaviz#665]
Jupyter lab has two built-in themes - the default "light" mode, and a "dark mode" that you have to switch into.

For context, when I use the default theme, specviz looks normal and fine the way we've intended: 
![image](https://user-images.githubusercontent.com/346587/121212448-5ccd6880-c84b-11eb-9d9f-33cd6873de44.png)

In dark mode, however, this is what I see:
![image](https://user-images.githubusercontent.com/346587/121212501-6c4cb180-c84b-11eb-8db1-e7b2498391f8.png)

You can see that it's stylistically a bit odd (not necessarily "wrong", but hard-to-read in some ways), but also the axis tick labels are missing, and the toolbar is shifted off to the right off the edge of the tool :zany_face: .

So I see two sets of fixes here - those that are *necessary* and a more "nice to have" level.

Should fix:
* change the white part of the jdaviz background to whatever the theme color should be for the background.  That will make the orange text of the axis labels much more readable and also better respect the user's expressed theme wish of a "dark mode"
* Get the axis tick labels back.  They might be there already and just be white for some reason, in which case maybe the fix above will do it?
* Fix whatever it is that makes the toolbar fall off the right side

Nice to have (@Jenneh might have some ideas here?):
* Change the "boundary line" color to be more appropriate for dark mode (right now it seems to be totally invisible)
* Re-think the color scheme to be better for dark mode.  E.g., that orange color seems a bit odd.
* Adjust some of the fonts to be more appropriate for light-on-dark.

**Comment by Jenneh** [comment]
- I agree that the most important thing will be to set the white background to black. (perhaps for the whole webpage..?)
- I suspect the tic labels are white, and if so should can remain white on the black background.
- Set the labels (wavelength , angstrom...) to white rather than yellow.

**Comment by pllim** [comment]
Is dark mode the only theme we have to worry about or are there other color schemes?

**Comment by pllim** [comment]
Might be related: #415

**Comment by eteq** [comment]
re: 

> Is dark mode the only theme we have to worry about or are there other color schemes?

The only ones I see are the the "light" and "dark" themes.  Note this is the *Jupyterlab* themes, not jdaviz-provided. I've not heard anything additional themes planned, but :shrug: I also didn't know about dark mode until I found it in the menu one day (and haven't looked back since until this issue!)

**Comment by eteq** [comment]
Oh and re: @Jenneh's

> (perhaps for the whole webpage..?)

That part is already set - I probably should have given a whole-page view:
![image](https://user-images.githubusercontent.com/346587/121233716-e8052900-c860-11eb-8d09-b7820b1f6440.png)

**Comment by maartenbreddels** [comment]
@mariobuikhuizen i guess ipyvuetify doesn't respond to the theme change of jlab right?

The white around the bqplot figure is probably an ipywidget thing, or ipywidget inside of vue that causes the issue. I think also here @mariobuikhuizen would be the person to solve this.

**Comment by mariobuikhuizen** [comment]
You can set the ipyvuetify dark mode with:
```python
import ipyvuetify
ipyvuetify.theme.dark = True
```
Ideally, I think ipyvuetify should do this automatically in a Jupyterlab context. I will take a look at that.

**Comment by mariobuikhuizen** [comment]
Version 1.7.0 of ipyvuetify now automatically adapts to light/dark theme settings in Jupyter Lab.

The toolbar falling off the side is a different issue. This also happens when making the browser window smaller.

**Comment by pllim** [comment]
> settings in Jupyter Lab

So if we're using normal notebook or `viola` app, this is simply not available?

**Comment by mariobuikhuizen** [comment]
The classic notebook doesn't have a theme, but you can set the vuetify theme to dark with 
```python
import ipyvuetify
ipyvuetify.theme.dark = True
```

In Voila the dark theme can be enabled by setting --theme=dark (not sure on how to pass this via cli.py), for which #682 adds support.

**Comment by mariobuikhuizen** [comment]
For the right side of the app falling out of view, I have found the following workaround (which changes the style of JupyterLab):
```diff
diff --git a/jdaviz/app.vue b/jdaviz/app.vue
index 54c9e53..ac33a1c 100644
--- a/jdaviz/app.vue
+++ b/jdaviz/app.vue
@@ -197,4 +197,8 @@ div.output_wrapper {
   padding: 0px;
   margin: 0px;
 }
+
+.jp-OutputArea-output {
+  overflow-y: hidden;
+}
 </style>
```

Not sure yet how to properly fix it.

**Comment by pllim** [comment]
> not sure on how to pass this via cli.py

Does the existing code around here give you hint on how to expose dark theme via CLI?

https://github.com/spacetelescope/jdaviz/blob/664d35ba6a5379bd838ae6466721c4123da5df57/jdaviz/cli.py#L28

**Comment by mariobuikhuizen** [comment]
I mistakenly thought #669 was an issue, but it's actually a PR. The code in it handles the setting of the dark theme via the CLI.  So #669 together with #682 is a complete fix for the dark theme in Voila. Maybe it's better to base my PR(#682) on #669.

**Comment by pllim** [comment]
Update: I merged #669 into #682 with my maintainer power. Hope you don't mind. Let's discuss over at #682.
